File: ITsupportBack/src/main/java/com/app/Model/Ticket.java
package com.app.Model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Data
public class Ticket {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long idTi;
    private String description;
    private LocalDate dateCréation;

    @Enumerated(EnumType.STRING)
    private EtatTicket etatTicket;
    @ManyToOne
    @JoinColumn(name = "idPanne", nullable = false)
    private Panne panne;

    @ManyToOne
    @JoinColumn(name = "idEquipement", nullable = false)
    private Equipement equipement;

    @ManyToOne
    @JoinColumn(name = "iduser", nullable = false)
    private Utilisateur utilisateur;

    @ManyToOne
    @JoinColumn(name = "idtechnicien", nullable = false)
    private TechnicienIT technicien;
}

File: ITsupportFront/src/app/Module/ticket.ts
export interface Ticket {
    id?: number;
    equipement: { idEquipement: number };
    panne: { idPanne: number };
    utilisateur?: { id: number };
    etatTicket?: EtatTicket;
    dateCreation?: string;
    description?: string;  // Ajout du champ description
  }
  
  
  export enum EtatTicket {
    OUVERT = 'OUVERT',
    EN_COURS = 'EN_COURS',
    EN_ATTENTE = 'EN_ATTENTE',
    FERME = 'FERME'
  }